jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/TreeToPathDecompositionAlgorithm.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * An algorithm which computes a decomposition into disjoint paths for a given tree/forest
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public interface TreeToPathDecompositionAlgorithm<V, E> {
+    /**
+     * Computes a path decomposition.
+     *
+     * @return a path decomposition
+     */
+    PathDecomposition<V, E> getPathDecomposition();
+
+    /**
+     * A path decomposition.
+     *
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     */
+    interface PathDecomposition<V, E> {
+        /**
+         * Set of edges of the path decomposition.
+         * 
+         * @return edge set of the path decomposition
+         */
+        Set<E> getEdges();
+
+        /**
+         * List of disjoint vertex paths of the decomposition
+         *
+         * @return list of vertex paths
+         */
+        List<List<V>> getPaths();
```

Comment:
Is there a reason not to produce org.jgrapht.GraphPath<V,E> instead of List<V>?  org.jgrapht.graph.GraphWalk makes it easy to produce a GraphPath from a List<V>.

Also, technically it should be Set<GraphPath> rather than List<GraphPath>.  It's possible to do this efficiently with an ArrayUnenforcedSet (elements unique by construction) wrapped as unmodifiable.

Reply:
Yes, I forgot about `GraphPath`. :)

I've made that change but I don't think that `PathDecompositionImpl` should be `Serializable` anymore since `GraphWalk` isn't. Any thoughts on this?

Reply:
Seems like GraphWalk should be made serializable as well.

Reply:
I can submit a PR for that.

Reply:
#628 solves this 'issue'.

Reply:
`GraphWalk` contains a reference to the graph instance. Not sure it makes sense to make it serializable.

Reply:
A number of our other algorithm return-values (cycle basis, spanner, spanning tree) are serializable with graph references.  I can see how in other contexts, a user might be unpleasantly surprised to find the entire graph serialized after they asked to only serialize a small walk by itself, but they'll get over it.  In the decomposition case, we're returning a bunch of paths all over the graph, and serialization takes care of only saving one copy of the graph automatically.